File: go/gigasecond/gigasecond.go
package gigasecond

import "time"

const gigasecond = 1e9 * time.Second

// AddGigasecond - Adds a gigasecond to a given time
func AddGigasecond(t time.Time) time.Time {
	return t.Add(gigasecond)
}

File: java/gigasecond/src/main/java/Gigasecond.java
import java.time.Duration;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

public class Gigasecond {
    private static final Duration gigasecond = Duration.ofSeconds(1_000_000_000);
    private final LocalDateTime gigamoment;

    public Gigasecond(LocalDate moment) {
        this(LocalDateTime.of(moment, LocalTime.MIN));
    }

    public Gigasecond(LocalDateTime moment) {
        this.gigamoment = moment.plus(gigasecond);
    }

    public LocalDateTime getDateTime() {
        return this.gigamoment;
    }
}
